src/metabase/driver/sql_jdbc.clj

```diff
@@ -148,6 +148,15 @@
                                :dialect (sql.qp/quote-style driver)))]
     (qp.writeback/execute-write-sql! db-id sql)))
 
+(defmethod driver/truncate! :sql-jdbc
+  [driver db-id table-name]
+  (let [table-name (keyword table-name)
+        sql        (sql/format {:truncate table-name}
+                               :quoted true
+                               :dialect (sql.qp/quote-style driver))]
+    (jdbc/with-db-transaction [conn (sql-jdbc.conn/db->pooled-connection-spec db-id)]
+      (jdbc/execute! conn sql))))
```

Comment:
Why aren't you using `qp.writeback/execute-write-sql!`? We reuse that function for permissions stuff

Reply:
Ok, well it seems to work inside the transaction so I've made the switch.

Reply:
Are you not talking about the checks in `upload/can-update-error`?

Reply:
Yeah it occurs to me now that maybe we first used `qp.writeback/execute-write-sql!` to do permissions checks for uploads and then later I added `upload/can-update-error` to cover appends, which is why `insert-into!` never used it.

Reply:
nope, I just checked and seems we always did the `perms/set-has-full-permissions?` check. But Noah definitely changed something about how native write perms work with block permissions, so maybe that's why the tests pass now.

If that's the case then we can remove this 
https://github.com/metabase/metabase/blob/739c02205a03451e58b3c2199da62dedf2d20fef/src/metabase/upload.clj#L555-L557 

and remove our usage of `api/*current-user*` from the uploads namespace, replacing it with an argument. Something for another PR though.